FreeRTOS/FreeRTOS-Plus-TCP
# Repair FreeRTOS_SendPingRequest

Description
-----------
`FreeRTOS_SendPingRequest()` first allocates a network buffer and then it does the sanity checks like length and the number of available network buffers. Under some circumstances, this could lead to the loss of network buffers.
The order should be the other way around: check if the ICMP/ping packet may be sent, and then allocate the buffer.

Test Steps
-----------
For instance, call `FreeRTOS_SendPingRequest()` with `uxNumberOfBytesToSend` set to zero.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
